### PR TITLE
[dev-v2.7] Bump chart cloud-provider-harvester to v0.2.3

### PR DESCRIPTION
**Pull request by w13915984028**
## Issue: 

https://github.com/harvester/harvester/issues/4703

## Problem

Harvester chart cloud-provider-harvester v0.2.3 (https://github.com/harvester/charts/releases/tag/harvester-cloud-provider-0.2.3) is released, but not in Rancher charts v2.7 and v2.8.

BTW, the `up0.2.3` fixed an important issue: https://github.com/harvester/harvester/issues/4678.

## Solution

Bump chart cloud-provider-harvester to v0.2.3

### CODE REVIEW

**Review by FrankYang0529 — APPROVED** [review]
LGTM.

### Test steps:

1. Start a Rancher.
2. Go to `Repositories` page and update the branch of `Rancher` repository to `pull/3234/head`.
Screenshot 2023-11-28 at 11 14 39 AM

3. Click `Refresh`.
Screenshot 2023-11-28 at 11 16 19 AM

4. After `Rancher` repository is active again, go to `Charts` page and click `Harvester Cloud Provider`. Check there is a new `Chart Versions`.
Screenshot 2023-11-28 at 11 17 09 AM

**Review by starbops — APPROVED** [review]
LGTM, thanks!

**Comment by w13915984028** [comment]
The cloud-provider-harvester chart is updated from v0.2.2 to v0.2.3 due to a bug fix. We therefore rebump it to rancher.

A test similar to above steps, and the chart version is shown as:
image

File: charts/harvester-cloud-provider/102.0.2+up0.2.3/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+annotations:
+  catalog.cattle.io/certified: rancher
+  catalog.cattle.io/display-name: Harvester Cloud Provider
+  catalog.cattle.io/kube-version: '>= 1.23.0-0 < 1.27.0-0'
```
Comment (bk201) [inline charts/harvester-cloud-provider/102.0.2+up0.2.3/Chart.yaml]: @w13915984028 Rancher team is going to support v1.27 in 2.7, so we need to bump the condition here.
See https://github.com/rancher/rancher/issues/44126 for the new issue.

**Review by lucasmlp — APPROVED** [review]
LGTM 🚀

**Pull request merged**